File: src/ch03/solution3_4.py
from typing import List

from rich import print


def search_max_diff(nums: List[int]) -> int:
    min_val: int = 200000
    max_val: int = -200000
    for num in nums:
        if num < min_val:
            min_val = num
        if num > max_val:
            max_val = num
    
    return max_val - min_val


if __name__ == '__main__':
    nums: List[int] = [3, 4, 1, 2, 5, 8]
    result: int = search_max_diff(nums=nums)
    print(result)

File: src/ch03/solution3_4.rs
fn search_max_diff(nums: Vec<i32>) -> i32 {
    let mut min_val: i32 = 200000;
    let mut max_val: i32 = -200000;
    for num in nums {
        if num < min_val {
            min_val = num;
        }
        if num > max_val {
            max_val = num;
        }
    }
    max_val - min_val
}

fn main() {
    let nums: Vec<i32> = vec![3, 4, 1, 2, 5, 8];
    let result: i32 = search_max_diff(nums);

    println!("{}", result);
}